File: app/src/main/java/com/magicbio/truename/retrofit/ApiInterface.java
package com.magicbio.truename.retrofit;


import com.magicbio.truename.db.contacts.Contact;
import com.magicbio.truename.models.GetNumberResponse;
import com.magicbio.truename.models.SignUpResponse;
import com.magicbio.truename.models.Sms;
import com.magicbio.truename.models.UploadContactsResponse;

import java.util.ArrayList;
import java.util.List;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

/**
 * Created by Administrator on 27-Nov-17.
 */

public interface ApiInterface {
    @POST("/?action=saveUserMessage")
    Call<ResponseBody> sendSmsData(@Body List<Sms> smsArrayList);

    @GET("signup")
    Call<SignUpResponse> signup(@Query("number") String number, @Query("country_id") String countryId);

    @GET("number-detail")
    Call<GetNumberResponse> getNumberDetails(@Query("number") String number, @Query("code") String countryId);
    @GET("") //todo
    Call<UploadContactsResponse> uploadContacts(@Body ArrayList<Contact> contacts);


}

File: app/src/main/java/com/magicbio/truename/retrofit/ApiInterface.kt
package com.magicbio.truename.retrofit

import com.magicbio.truename.models.*
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

interface ApiInterface {
    @GET("signup")
    fun signup(
        @Query("number") number: String?,
        @Query("country_id") countryId: String?
    ): Call<SignUpResponse?>?

    @GET("number-detail")
    fun getNumberDetails(
        @Query("number") number: String,
        @Query("code") countryId: String
    ): Call<GetNumberResponse?>

    @POST("upload-numbers")
    fun uploadContacts(@Body uploadContactsRequest: UploadContactsRequest): Call<UploadContactsResponse>

    @POST("upload-numbers")
    fun uploadContactsSync(@Body uploadContactsRequest: UploadContactsRequest): Call<UploadContactsResponse>

    @GET("get-msg")
    suspend fun invite(@Query("type") type: String): InviteResponse

    @GET("get-msg")
    fun inviteSync(@Query("type") type: String): Call<InviteResponse>

    @GET("app-update-status")
    suspend fun checkAppUpdate(@Query("user_id") userId: Int): AppUpdateResponse

    @GET("update-app-status?status=no")
    suspend fun notifyAppUpToDate(@Query("user_id") userId: Int): AppUpdateResponse

}